go/controller/internal/controller/servicedeployment_controller.go

```diff
@@ -143,7 +145,6 @@ func (r *ServiceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_
 			utils.MarkCondition(service.SetCondition, v1alpha1.SynchronizedConditionType, v1.ConditionFalse, v1alpha1.SynchronizedConditionReasonError, err.Error())
 			return ctrl.Result{}, err
 		}
-		controllerutil.AddFinalizer(service, ServiceFinalizer)
```

Comment:
I got distracted and haven't finished second comment. There is also slight chance that finalizer was not added after resource creation. Now it's added before resource is created which should also make it more error proof.